File: Deployment_Mode/LoglevelGrouping.py
import sys
from pyspark import SparkContext

def main(logFile):

	#Create SparkContext object
    sc = SparkContext()
    
    #Read the file from cluster
    rdd1 = sc.textFile(logFile)
    
    #Find the Loglevel Grouping
    rdd2 = rdd1.map(lambda x: (x.split(":")[0], x.split(":")[1])).groupByKey().map(lambda x: (x[0], len(x[1])))
    
    #Perform action to print the number of entries
    rdd3 = rdd2.collect()
    
    #Display the results
    for k, v in rdd3:
        print('Key:{k}, Value:{v}'.format(k=k, v=v))
    
    
if __name__ == "__main__":
    args = sys.argv[1]
    #print(args)
    main(args)

File: Deployment_Mode/LoglevelGrouping.scala
import org.apache.spark.SparkContext
import org.apache.log4j._

object LoglevelGrouping {
  
  //Define main method
  def main(args: Array[String]){
    
    //Set log Level
    Logger.getLogger("org").setLevel(Level.WARN)
    //Create a SparkContext
    val sc = new SparkContext()
    
    //Read Input file with the command line argument as the input file location
    val rdd1 = sc.textFile(args(0))
    
    val rdd2 = rdd1.map(x=>(x.split(":")(0), x.split(":")(1))).groupByKey.map(x=>(x._1, x._2.size))
    
    rdd2.collect().foreach(println)
    
  }
  
}